pennylane/labs/resource_estimation/ops/qubit/non_parametric_ops.py

```diff
@@ -0,0 +1,612 @@
+
+        Raises:
+            ResourcesNotDefined: This gate is fundamental, no further decomposition defined.
+        """
+        raise plre.ResourcesNotDefined
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: Empty dictionary. The resources of this operation don't depend on any additional parameters.
+        """
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute the resources."""
+        return CompressedResourceOp(cls, {})
+
+    @classmethod
+    def default_adjoint_resource_decomp(cls) -> list[GateCount]:
+        r"""Returns a list representing the resources for the adjoint of the operator.
+
+        Resources:
+            The adjoint of the T-gate is equivalent to the T-gate raised to the 7th power.
+            The resources are defined as one Z-gate (:math:`Z = T^{4}`), one S-gate (:math:`S = T^{2}`) and one T-gate.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+                represents a specific quantum gate and the number of times it appears
+                in the decomposition.
+        """
+        z = resource_rep(ResourceZ)
+        s = resource_rep(ResourceS)
+        return [GateCount(cls.resource_rep()), GateCount(s), GateCount(z)]
+
+    @classmethod
+    def default_pow_resource_decomp(cls, pow_z) -> list[GateCount]:
+        r"""Returns a list representing the resources for an operator raised to a power.
+
+        Args:
+            pow_z (int): the power that the operator is being raised to
+
+        Resources:
+            The T-gate, when raised to a power which is a multiple of eight, produces identity.
+            Consequently, for any integer power `z`, the effective quantum operation :math:`T^{z}` is equivalent
+            to :math:`T^{z \pmod 8}`.
+
+        The decomposition for :math:`T^{z}` (where :math:`z \pmod 8` is denoted as `z'`) is as follows:
+            - If `z' = 0`: The operation is equivalent to the Identity gate (`I`).
+            - If `z' = 1`: The operation is equivalent to the T-gate (`T`).
+            - If `z' = 2`: The operation is equivalent to the S-gate (`S`).
+            - If `z' = 3`: The operation is equivalent to a composition of an S-gate and a T-gate (:math:`S \cdot T`).
+            - If `z' = 4` : The operation is equivalent to the Z-gate (`Z`).
+            - If `z' = 5`: The operation is equivalent to a composition of a Z-gate and a T-gate (:math:`Z \cdot T`).
+            - If `z' = 6`: The operation is equivalent to a composition of a Z-gate and an S-gate (:math:`Z \cdot S`).
+            - If `z' = 7`: The operation is equivalent to a composition of a Z-gate, an S-gate and a T-gate.
```

Comment:
This is not rendering well either. I think we need to indent it to match the above.
```suggestion
            The decomposition for :math:`T^{z}` (where :math:`z \pmod 8` is denoted as `z'`) is as follows:
                - If `z' = 0`: The operation is equivalent to the Identity gate (`I`).
                - If `z' = 1`: The operation is equivalent to the T-gate (`T`).
                - If `z' = 2`: The operation is equivalent to the S-gate (`S`).
                - If `z' = 3`: The operation is equivalent to a composition of an S-gate and a T-gate (:math:`S \cdot T`).
                - If `z' = 4` : The operation is equivalent to the Z-gate (`Z`).
                - If `z' = 5`: The operation is equivalent to a composition of a Z-gate and a T-gate (:math:`Z \cdot T`).
                - If `z' = 6`: The operation is equivalent to a composition of a Z-gate and an S-gate (:math:`Z \cdot S`).
                - If `z' = 7`: The operation is equivalent to a composition of a Z-gate, an S-gate and a T-gate.
```